blogs/ecs-ec2-crossvpc-with-tls/appmesh-baseline.yml

```diff
@@ -0,0 +1,937 @@
+---
+AWSTemplateFormatVersion: "2010-09-09"
+Description: This stack deploys the baseline infrastructure
+             to be used in the App Mesh Workshop.
+
+Parameters:
+
+  Cloud9IAMRole: 
+    Type: String
+    Default: AppMesh-Workshop-Admin
+
+  LatestAmiId:
+    Type: 'AWS::SSM::Parameter::Value<AWS::EC2::Image::Id>'
+    Default: '/aws/service/ami-amazon-linux-latest/amzn2-ami-hvm-x86_64-gp2'
+
+Mappings:
+
+  SubnetConfig:
+    VPC:
+      CIDR: '10.0.0.0/16'
+    PublicOne:
+      CIDR: '10.0.0.0/24'
+    PublicTwo:
+      CIDR: '10.0.1.0/24'
+    PublicThree:
+      CIDR: '10.0.2.0/24'
+    PrivateOne:
+      CIDR: '10.0.100.0/24'
+    PrivateTwo:
+      CIDR: '10.0.101.0/24'
+    PrivateThree:
+      CIDR: '10.0.102.0/24'
+
+Resources:
+        
+  Crystal:
+    Type: AWS::ECR::Repository
+
+  NodeJS:
+    Type: AWS::ECR::Repository
+
+  ECSCluster:
+    Type: AWS::ECS::Cluster
+    Properties:
+      ClusterName: !Sub cluster-${AWS::StackName}
+
+  CrystalTaskDefinition:
+    Type: AWS::ECS::TaskDefinition
+    Properties: 
+      Cpu: 256
+      ExecutionRoleArn: !GetAtt ECSServiceRole.Arn
+      Family: !Sub crystal-task-${AWS::StackName} 
+      Memory: 512
+      NetworkMode: awsvpc
+      RequiresCompatibilities:
+        - FARGATE
+      TaskRoleArn: !GetAtt ECSTaskRole.Arn
+      ContainerDefinitions:
+        - Name: crystal-service
+          Image: brentley/ecsdemo-crystal
+          PortMappings:
+            - ContainerPort: 3000
+              Protocol: http
+          Essential: true
+          DependsOn:
+            - ContainerName: "envoy"
+              Condition: "HEALTHY"
+          HealthCheck:
+            Command: 
+              - CMD-SHELL
+              - curl -s http://localhost:3000/health | grep -q Healthy!
+            Interval: 5
+            Retries: 3
+            StartPeriod: 10
+            Timeout: 2
+        - Name: envoy
+          Image: "840364872350.dkr.ecr.eu-west-2.amazonaws.com/aws-appmesh-envoy:v1.15.1.0-prod"
```

Comment:
Could this be a parameter that's passed in? We should instruct customers to use the latest envoy image from https://docs.aws.amazon.com/app-mesh/latest/userguide/envoy.html.

Reply:
Introduced EnvoyImage parameter.